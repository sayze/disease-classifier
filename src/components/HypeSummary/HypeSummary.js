import React from 'react'
import { Typography } from '@material-ui/core'
import PropTypes from 'prop-types'
import Stage from 'services/Hypertension'

const HypeSummary = ({ date, sysBp, dialBp, stage }) => (
  <>
    <Typography component="span">{date}</Typography>
    <Typography component="span">{sysBp}</Typography>
    <Typography component="span">{dialBp}</Typography>
    <Typography component="span">{stage.toString()}</Typography>
  </>
)

HypeSummary.defaultProps = {
  date: new Date().toDateString(),
  sysBp: 0,
  dialBp: 0,
}

HypeSummary.propTypes = {
  date: PropTypes.string,
  sysBp: PropTypes.number,
  dialBp: PropTypes.number,
  stage: PropTypes.instanceOf(Stage).isRequired,
}

export default HypeSummary
