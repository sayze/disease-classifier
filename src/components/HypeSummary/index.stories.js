import React from 'react'
import HypeSummary from '.'
import { withKnobs, number } from '@storybook/addon-knobs'

export default {
  title: 'HyperSummary',
  component: HypeSummary,
  decorators: [withKnobs],
}
export const Default = () => {
  const dialBp = number('Diastolic Blood Pressure', 0)
  const sysBp = number('Systolic Blood Pressure', 0)
  return <HypeSummary date="21/05/2020" dialBp={dialBp} sysBp={sysBp} />
}
