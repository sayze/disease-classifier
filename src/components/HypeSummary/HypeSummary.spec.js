import React from 'react'
import { shallow } from 'enzyme'
import HypeSummary from '.'

describe('HypeSummary', () => {
  it('should render a valid hypertension summary', () => {
    const component = shallow(<HypeSummary date={new Date('2020-01-01').toDateString()} />)
    expect(component).toMatchSnapshot()
  })
})
