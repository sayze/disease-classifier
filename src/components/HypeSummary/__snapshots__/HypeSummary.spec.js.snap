// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HypeSummary should render a valid hypertension summary 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      Reading Date: 
      Wed Jan 01 2020
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      Systolic Blood Pressure: 
      0
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    >
      Diastolic Blood Pressure: 
      0
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
